hivemind/averaging/allreduce.py

```diff
@@ -224,9 +231,9 @@ async def _accumulate_parts_streaming(self, stream: AsyncIterator[averaging_pb2.
             yield averaging_pb2.AveragingData(code=averaging_pb2.AVERAGED_PART, tensor_part=serialized_delta)
 
     async def _send_error_to_peer(self, peer_endpoint: Endpoint, code: averaging_pb2.MessageCode):
-        stream = self._get_peer_stub(peer_endpoint).rpc_aggregate_part()
-        await stream.write(averaging_pb2.AveragingData(group_id=self.group_id, endpoint=self.endpoint, code=code))
-        await stream.done_writing()
+        error = averaging_pb2.AveragingData(group_id=self.group_id, endpoint=self.endpoint.to_base58(), code=code)
+        async for _ in self._get_peer_stub(peer_endpoint).rpc_aggregate_part(aiter(error)):
+            pass
```

Comment:
Maybe a comment with an explanation of why we don't make an unary call here would be helpful

Reply:
We can't use a unary call here just because the function has a `stream -> stream` signature.

Nevertheless, I've made the code more readable with the new `asingle()` function and added a comment.